tools/universe/package.py

```diff
@@ -0,0 +1,34 @@
+from .package_manager import PackageManager
+from .version_resolver import VersionResolver
+
+
+class Package:
```

Comment:
Now that I think about it, how about moving the version lookup outside of the Package class, and having the Package class function as a data-only struct/container for a name+version?

- `PackageManager`/`VersionResolver` could accept a package name as a parameter and return a Package object.
- `_get_template_mapping_for_content()` could deal with lookup and `"*"` fallback internally
- `PackageManager`'s cache would be more useful since it would survive across package names by default. It could hold a pre-parsed dict of `name => Package` too. Debug logs could print the list of package versions that it found by dumping that dict.

(This comment was written after the following comments)

Reply:
Also, I could see us wanting to have deeper relationships between `Package` objects at some point, for example having a `Package` object graph that follows the `upgradesFrom`/`downgradesTo` paths. Having `PackageManager` construct the `Package`s directly could make constructing this graph a little easier.